src/Semantic/Util.hs

```diff
@@ -76,54 +64,29 @@ justEvaluating
-                (ModuleTable (Module (ModuleResult Precise (Value (Quieterm (Sum syntax) Loc) Precise))))))
-
-evalGoProject :: FileEvaluator _ Language.Go.Assignment.Syntax
-evalGoProject = justEvaluating <=< evaluateProject (Proxy :: Proxy 'Language.Go) goParser
-
-evalRubyProject :: FileEvaluator _ Language.Ruby.Assignment.Syntax
-evalRubyProject = justEvaluating <=< evaluateProject (Proxy @'Language.Ruby)               rubyParser
-
-evalPHPProject :: FileEvaluator _ Language.PHP.Assignment.Syntax
-evalPHPProject  = justEvaluating <=< evaluateProject (Proxy :: Proxy 'Language.PHP)        phpParser
-
-evalPythonProject :: FileEvaluator _ Language.Python.Assignment.Syntax
-evalPythonProject = justEvaluating <=< evaluateProject (Proxy :: Proxy 'Language.Python)     pythonParser
-
-evalTypeScriptProject :: FileEvaluator _ Language.TypeScript.Assignment.Syntax
-evalTypeScriptProject = justEvaluating <=< evaluateProject (Proxy :: Proxy 'Language.TypeScript) typescriptParser
-
-evaluateProject proxy parser paths = withOptions debugOptions $ \ config logger statter ->
-  evaluateProject' (TaskSession config "-" False logger statter) proxy parser paths
-
 -- Evaluate a project consisting of the listed paths.
 evaluateProject' session proxy parser paths = do
+  let lang = Language.reflect proxy
   res <- runTask session $ asks configTreeSitterParseTimeout >>= \ timeout -> runParse timeout $ do
-    blobs <- catMaybes <$> traverse readBlobFromFile (flip File (Language.reflect proxy) <$> paths)
-    package <- fmap (quieterm . snd) <$> parsePackage parser (Project (takeDirectory (maybe "/" fst (uncons paths))) blobs (Language.reflect proxy) [])
+    blobs <- catMaybes <$> traverse readBlobFromFile (flip File lang <$> paths)
+    package <- fmap snd <$> parsePackage parser (Project (takeDirectory (maybe "/" fst (uncons paths))) blobs lang [])
     modules <- topologicalSort <$> runImportGraphToModules proxy package
     trace $ "evaluating with load order: " <> show (map (modulePath . moduleInfo) modules)
-    pure (id @(Evaluator _ Precise (Value _ Precise) _ _)
-         (runModuleTable
-         (runModules (ModuleTable.modulePaths (packageModules package))
-         (raiseHandler (runReader (packageInfo package))
-         (raiseHandler (evalState (lowerBound @Span))
-         (raiseHandler (runReader (lowerBound @Span))
-         (evaluate proxy (runDomainEffects (evalTerm (withTermSpans (Loc.span . termFAnnotation . project)))) modules)))))))
-  either (die . displayException) pure res
+    pure (package, modules)
+  (package, modules) <- either (die . displayException) pure res
+  pure (runModuleTable
+       (runModules (ModuleTable.modulePaths (packageModules package))
+       (raiseHandler (runReader (packageInfo package))
+       (raiseHandler (evalState (lowerBound @Span))
+       (raiseHandler (runReader (lowerBound @Span))
+       (evaluate proxy (runDomainEffects (evalTerm (withTermSpans (^. span_)))) modules))))))
```

Comment:
This is mostly just rearranging things; I was trying to get to the point where I could abstract over the parser here, but it didn’t really work out, because this function returns values whose type is parameterized by the term type.

I even tried existentially quantifying over that, but that didn’t work out either because then we wouldn’t be able to equate them in the tests.